src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -57,6 +65,22 @@ public virtual void ApplyRule(RewriteContext context)
                 }
             }
 
+            foreach (ServerVariable serverVariable in ServerVariables)
+            {
+                if (serverVariable.Name.StartsWith(RequestHeaderPrefix, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
i did migrate them into a constants file but not following your suggestion here. can you clarify?